src/ImageSharp/Formats/Jpeg/PdfJsPort/Components/PdfJsScanDecoder.cs

```diff
@@ -201,6 +203,12 @@ internal struct PdfJsScanDecoder
             if (componentsLength == 1)
             {
                 PdfJsFrameComponent component = components[this.compIndex];
+
+                // TODO: This is where our error is happening.
+                // We can't simply cast the span as I think the scan decoder expects data to be laid out in linear order
+                // rather than in the column major order expected by the Block8x8 struct and anything reading it down the pipeline.
+                // Ask Anton about this. It might be a lost cause.
+                ref short blockDataRef = ref MemoryMarshal.GetReference(MemoryMarshal.Cast<Block8x8, short>(component.SpectralBlocks.Span));
```

Comment:
I suggest to to store the "reference" to the **current** block memory area as a `Span<short>` member right into the `PdfJsScanDecoder` struct. It would eliminate some of the repetitive code. We can extract the reference in methods like `DecodeACFirst`. I don't expect noticable performance regression from this change, but the code will be cleaner:
- No need to pass `blockDataRef` as method argument
- No need to pass `offset`